lib/client.ex

```diff
@@ -0,0 +1,153 @@
+      "script-format": Dict.get(opts, :scriptformat),
+      "offset": Dict.get(opts, :offset),
+      "limit": Dict.get(opts, :limit),
+    })
+    query = URI.encode_query(attrs)
+    req_url = add_query(url("/kapacitor/v1/tasks", conf[:url]), query)
+    request :get, req_url, [], "", ""
+  end
+
+  def define_task(conf, type, dbrps, script, status \\ :enabled, vars \\ %{}, id \\ nil, template_id \\ nil) do
+    do_define_task(conf, type, dbrps, script, status, vars, id, template_id)
+  end
+
+  defp do_define_task(conf, type, dbrps, script, status, vars, id , template_id) do
+    attrs = %{
+      "type": type,
+      "dbrps": dbrps,
+      "script": script,
+      "status": status,
+      "vars": vars,
+      "id": id,
+      "template_id": template_id,
+    }
+    req_url = url("/kapacitor/v1/tasks", conf[:url])
+
+    body = Poison.encode!(attrs)
+    request :post, req_url, [], "application/json", body
+  end
+
+  def enable_task(conf, id) do
+    do_update_task_status(conf, id, "enabled")
+  end
+
+  def disable_task(conf, id) do
+    do_update_task_status(conf, id, "disabled")
+  end
+
+  defp do_update_task_status(conf, id, status) do
+    attrs = %{
+      "status": status,
+    }
+    req_url = url(Path.join("/kapacitor/v1/tasks/", id), conf[:url])
+
+    body = Poison.encode!(attrs)
+    request :patch, req_url, [], "application/json", body
+  end
+
+  def delete_task(conf, id) do
+    do_delete_task(conf, id)
+  end
+
+  defp do_delete_task(conf, id) do
+    req_url = url(Path.join("/kapacitor/v1/tasks/", id), conf[:url])
+
+    request :delete, req_url, [], "", ""
+  end
+
+  def get_task(conf, id, dot_view \\ "attributes", script_format \\ "formatted") do
+    do_get_task(conf, id, dot_view, script_format)
+  end
+
+  defp do_get_task(conf, id, dot_view, script_format) do
+    attrs = remove_empty_values(%{
+      "dot-view": dot_view,
+      "script-format": script_format,
+    })
+    query = URI.encode_query(attrs)
+    req_url = add_query(url(Path.join("/kapacitor/v1/tasks/", id), conf[:url]), query)
+    request :get, req_url, [], "", ""
+  end
+
+  def remove_empty_values(map) do
+    map
+      |> Enum.filter(fn {_, v} -> v != nil && v != "" end)
+      |> Enum.into(%{})
+  end
+
+  def url(path, domain), do: Path.join([domain, path])
+
+  def request(method, url, headers, ctype, body) do
+    url  = String.to_char_list(url)
+
+    case method do
+      :get ->
+        HTTPoison.get(url)
+      _    ->
+        headers = headers ++ [{'Content-Type', ctype}]
+        HTTPoison.request(
+          method,
+          url,
+          body,
+          headers
+        )
+    end
+    |> normalize_response
+  end
+
+ defp normalize_response(response) do
+    case response do
+      {:ok, %HTTPoison.Response{body: body, status_code: 200}} ->
+        {:ok, Poison.decode!(body) |> Enum.map(fn({k, v}) -> {String.to_atom(k), v} end) }
+      {:ok, %HTTPoison.Response{status_code: 204}} ->
+        {:ok}
```

Comment:
should probably return a body or something here, otherwise the users will need to match on this case as well as {:ok, ""}, {:error, ""}